table.cpp

```diff
@@ -12,10 +12,10 @@ using std::string;
 
 table_t::table_t(const string& conninfo, const string& name, const string& type, const columns_t& columns, const hstores_t& hstore_columns,
     const int srid, const int scale, const bool append, const bool slim, const bool drop_temp, const int hstore_mode,
-    const bool enable_hstore_index, const boost::optional<string>& table_space, const boost::optional<string>& table_space_index) :
+    const bool enable_hstore_index, const boost::optional<string>& table_space, const boost::optional<string>& table_space_index, const boost::optional<string>& schema) :
```

Comment:
Jep, having a table_options_t sounds like the most clean solution. Maybe something for another pull request.